internal/fwserver/server_planresourcechange_test.go

```diff
@@ -1935,6 +1976,76 @@ func TestServerPlanResourceChange(t *testing.T) {
 				PlannedPrivate: testEmptyPrivate,
 			},
 		},
+		"update-resourcewithmodifyplan-request-config-nil-block": {
+			server: &fwserver.Server{
+				Provider: &testprovider.Provider{},
+			},
+			request: &fwserver.PlanResourceChangeRequest{
+				Config: &tfsdk.Config{
+					Raw: tftypes.NewValue(testSchemaBlockType, map[string]tftypes.Value{
+						"test_required": tftypes.NewValue(tftypes.String, "test-new-value"),
+						"test_optional_block": tftypes.NewValue(tftypes.Object{
+							AttributeTypes: map[string]tftypes.Type{
+								"test_optional_one": tftypes.String,
+								"test_optional_two": tftypes.String,
+							},
+						}, nil),
+					}),
+					Schema: testSchemaBlock,
+				},
+				ProposedNewState: &tfsdk.Plan{
+					Raw: tftypes.NewValue(testSchemaBlockType, map[string]tftypes.Value{
+						"test_required": tftypes.NewValue(tftypes.String, "test-new-value"),
+						"test_optional_block": tftypes.NewValue(tftypes.Object{
+							AttributeTypes: map[string]tftypes.Type{
+								"test_optional_one": tftypes.String,
+								"test_optional_two": tftypes.String,
+							},
+						}, nil),
+					}),
+					Schema: testSchemaBlock,
+				},
+				PriorState: &tfsdk.State{
+					Raw: tftypes.NewValue(testSchemaBlockType, map[string]tftypes.Value{
+						"test_required": tftypes.NewValue(tftypes.String, "test-new-value"),
```

Comment:
To trigger this new unit test to fail without the fix and therefore verify the fix, there needs to be a difference between the prior state and proposed new state to trigger the call to `MarkComputedNilsAsUnknown`. Adjusting this line should do the trick.

```suggestion
						"test_required": tftypes.NewValue(tftypes.String, "test-value"),
```

The reason this issue occurs upstream is mainly because during resource creation, the prior state is null, but the proposed new state contains blocks.